### PR TITLE
chore: ignore more GitHub tests requests

### PR DESCRIPTION
**Pull request by patheard**
# Summary 
Update the CloudWatch alarm and alert compromise to ignore more GitHub test request patterns.

Add unit tests, linting and formatting for the alert compromise function.

### CODE REVIEW

File: terragrunt/aws/alarms/locals.tf [hunk 1]
```diff
@@ -3,4 +3,14 @@ locals {
...
   metric_namespace         = "GitHubSecretScanning"
   secret_detected_api      = "SecretDetectedAPI"
+
+  secret_detected_metric = [
+    "Secret detected",
+  ]
+  secret_detected_metric_skip = [
+    "dsp-testing",
+    "example.com",
+    "gcntfy-github-test-revoked",
```
Comment (sylviamclaughlin) [inline terragrunt/aws/alarms/locals.tf:13]: Nice that you added the `gcntfy-github-test-revoled` one as well!

File: terragrunt/aws/alert_compromise/functions/broadcast_alert_tests.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
+import pytest
+from unittest import mock
+from broadcast_alert import lambda_handler
+
+
+@mock.patch("broadcast_alert.boto3.client")
```
Comment (sylviamclaughlin) [inline terragrunt/aws/alert_compromise/functions/broadcast_alert_tests.py:6]: Thank you for adding unit tests as well - I guess I should have probably done that before 😬

**Review by sylviamclaughlin — APPROVED** [review]
🎉

File: terragrunt/aws/alert_compromise/functions/broadcast_alert_tests.py (hunk 2 above)
Comment (patheard) [reply terragrunt/aws/alert_compromise/functions/broadcast_alert_tests.py:6]: All good!  Infinite work, finite time ⏳

**Pull request merged**